### ISSUE TITLE
sticking keys after registering global hotkeys

### ISSUE THREAD

**Issue by jfayre** [boppreh/keyboard#326]
Hello,
I'm writing an application that registers several global hotkeys. Some of my users are reporting that keys will suddenly stop responding, or some modifier keys become stuck. For example, Windows will think the ctrl key is held down but it isn't. 
I know the app needs to be running as admin for global keys to work. That isn't the issue.
This is python 3.7.6 and the latest version of the keyboard module from pip.  0.13.4.
The app is built using pyinstaller.

Any idea what could be going on? If it helps, I'm using suppress=True on all my key deffinitions.

**Comment by vikilpet** [comment]
I have the same problem with suppressed keys.
Those who are interested in Windows-only implementation of global hotkeys can look at the GlobalHotKeys class here:
https://github.com/vikilpet/Taskopy/blob/master/plugins/plugin_hotkey.py
There is a test example at the bottom.

**Comment by dpvdberg** [comment]
I have the same problem when using `suppress=True`

**Comment by ohMuzzy** [comment]
Hi Guys! Any luck with that? 

Just as a side note: unfortunately, the suggested GlobalHotKeys class, has a big downside to it, as it does not release ALT after pressing the final key. For example, on pressing C in ALT + C the alt key should be released, and it’s not the case here. This is not how Windows hotkeys behave, and it introduces multitude of problems.

Back to the main topic: I did some snooping around and know why the CTRL/ALT key gets stuck in the original Keyboard module:

It happens when you register a hotkey (with suppress=True) that includes CTRL or ALT, and then (intentionally or probably more often accidentally) press the RIGHT CTRL + ALT GR combination.
Here it is recreated. Just run it and press RIGHT CTRL + ALT GR combination, if you have ALT GR on your keyboard. (You will need to physically press LEFT CTRL a couple of times to restore normal keyboard operation afterwards.)

```
import keyboard

def on_hotkey(str):
    print(str)

keyboard.add_hotkey('ctrl+a', on_hotkey, args=['!'], suppress=True)

keyboard.wait()
```

Pressing RIGHT CTRL + ALT GR in this example creates the following sequence of events:

**LEFT** CTRL is pressed > ALT GR is pressed > ALT GR is released > **RIGHT** CTRL is released

leaving LEFT CTRL in permanently pressed position that cannot be released virtually (only by physically pressing the key a couple of times)

There is an analogous situation when you register an ALT-containing hotkey with LEFT ALT getting stuck.

How do you fix that?

**Comment by vikilpet** [comment]
ohMuzzy, i'm using *alt + ...*  hotkeys a lot (currently 8 tasks with *alt+* hotkey) and see no issue with sticked alt. `GlobalHotKeys` uses standard windows API.

**Comment by ohMuzzy** [comment]
Hi vikilpet!

The following code produces results in Notepad only if you press CTRL+T, not ALT+T.
If you try it in Word, none of the hotkeys work (no such issues with the Keyboard module).

It might be because of the modifiers being pressed at the same time as the function is executed.

If this is the case, is there a way to release all pressed modifs before executing the function and restore them afterwards?

```
if __name__ == '__main__':
    ''' Test this module: bind test_func to 'ctrl+t'
        global hotkey and exit test with 'ctrl+shift+t'
    '''
    import threading


    def test_func():
        keyboard.write('The quick brown fox jumps over the lazy dog')

    ghk = GlobalHotKeys()
    ghk.register('ctrl+t', func=test_func)
    ghk.register('alt+t', func=test_func)

    threading.Thread(target=ghk.listen).start()

else:
    patch_import()
```

**Comment by vikilpet** [comment]
> It might be because of the modifiers being pressed at the same time as the function is executed.
That's right. Alt is a modifier key. It moves the focus to the menu in many programs.
You can just add a pause  and it will work in Word an Notepad. Or use `release` and it will more or less work :)

    if __name__ == '__main__':
        ''' Test this module: bind test_func to 'ctrl+t'
            global hotkey and exit test with 'ctrl+shift+t'
        '''
        import threading
        import time

        def test_func():
            time.sleep(1)
            # keyboard.release('alt')
            keyboard.write('The quick brown fox jumps over the lazy dog')

        ghk = GlobalHotKeys()
        ghk.register('ctrl+t', func=test_func)
        ghk.register('alt+t', func=test_func)

        threading.Thread(target=ghk.listen).start()

**Comment by ohMuzzy** [comment]
Both options "work more or less".

The delay is a no no in my project plus it won't work if a user holds a key for longer than a second. Thanks for having a look anyway!

I'm still looking for the solution.
